src/PIL/EpsImagePlugin.py

```diff
@@ -333,22 +341,38 @@ def _read_comment(s: str) -> bool:
                 else:
                     break
 
-                self._size = columns, rows
-                return
+                # Read the columns and rows after checking the bit depth and mode
+                # in case the bit depth and/or mode are invalid.
+                imagedata_size = columns, rows
             elif bytes_mv[:5] == b"%%EOF":
                 break
             elif trailer_reached and reading_trailer_comments:
                 # Load EPS trailer
                 s = str(bytes_mv[:bytes_read], "latin-1")
-                _read_comment(s)
+                read_comment(s)
             elif bytes_mv[:9] == b"%%Trailer":
                 trailer_reached = True
             bytes_read = 0
 
-        if not self.tile:
+        # A "BoundingBox" is always required,
+        # even if an "ImageData" descriptor size exists.
+        if not bounding_box:
             msg = "cannot determine EPS bounding box"
             raise OSError(msg)
 
+        # An "ImageData" size takes precedence over the "BoundingBox".
```

Comment:
From my reading of https://github.com/python-pillow/Pillow/pull/8281#issuecomment-2267145829, is this logic just based on your experience of ImageMagick?

Reply:
Yes, and Photoshop also works this way: https://github.com/python-pillow/Pillow/issues/6382#issuecomment-1213597912

Reply:
Looking at https://paulbourke.net/dataformats/psdpsb/psdpsb.html, Photoshop actually originated this, so it does make some sense that Photoshop would prioritise their own comment.

> Photoshop includes a comment in the EPS files it writes so that it is able to read them back in again. Third party programs that write pixel-based EPS files may want to include this comment in their EPS files, so Photoshop can read their files.
> 
> The comment must follow immediately after the %% comment block at the start of the file. The comment is:
> 
> `%ImageData: <columns> <rows> <depth> <mode> <pad channels> <block size> <binary/hex> "<data start>"`